src/mpsc/blocking.rs

```diff
@@ -1165,6 +1313,37 @@ fn recv_ref<'a, T>(core: &'a ChannelCore<Thread>, slots: &'a [Slot<T>]) -> Optio
     }
 }
 
+#[inline]
+fn recv_ref_timeout<'a, T>(
+    core: &'a ChannelCore<Thread>,
+    slots: &'a [Slot<T>],
+    timeout: Duration,
```

Comment:
This function could be merged with the above by changing `Duration` to `Option<Duration>`, but it would require changing the error back to an option.

Reply:
hmm, i don't _love_ the duplication between `recv_ref` and `recv_ref_timeout`, but I think this is preferable to folding them together --- this method has additional code for taking a timestamp and checking if the timeout has elapsed, which it would be nice to avoid having in `recv_ref`, even if it's disabled by a check that no timeout was passed in. it's probably worthwhile to avoid the overhead of generating larger function bodies for the methods without timeouts...